File: src/main/java/hilfling/backend/hilfling/service/EventOwnerService.java
package hilfling.backend.hilfling.service;

import hilfling.backend.hilfling.model.EventOwner;
import hilfling.backend.hilfling.repositories.EventOwnerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class EventOwnerService extends
        GenericBaseServiceImplementation<EventOwner> {
    @Autowired
    EventOwnerRepository repository;
    public EventOwnerRepository getRepository() {
        return repository;
    }

}

File: src/main/kotlin/no/fg/hilflingbackend/service/EventOwnerService.kt
package no.fg.hilflingbackend.service

import jakarta.persistence.EntityNotFoundException
import no.fg.hilflingbackend.dto.EventOwnerDto
import no.fg.hilflingbackend.dto.Page
import no.fg.hilflingbackend.repository.EventOwnerRepository
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class EventOwnerService(
  val repository: EventOwnerRepository,
) {
  fun findById(id: UUID): EventOwnerDto = repository.findById(id) ?: throw EntityNotFoundException("EventOwner $id not found")

  fun findAll(
    page: Int,
    pageSize: Int,
  ): Page<EventOwnerDto> = repository.findAll(page, pageSize)

  fun create(dto: EventOwnerDto): Int = repository.create(dto)

  fun delete(id: UUID): Int = repository.delete(id)
}
